src/mqtt_client.rs

```diff
@@ -104,9 +114,15 @@ where
         match &packet {
             ReceivedPacket::SubAck(ack) => self.handle_suback(ack)?,
             ReceivedPacket::PingResp => self.session_state.register_ping_response(),
-            ReceivedPacket::PubComp(comp) => self.session_state.handle_pubcomp(comp.packet_id)?,
+            ReceivedPacket::PubComp(comp) => {
+                self.send_quota =
+                    core::cmp::min(self.send_quota.saturating_add(1), self.max_send_quota);
```

Comment:
```suggestion
                    self.send_quota.saturating_add(1).min(self.max_send_quota);
```

and in the two other places as well.

Reply:
I was trying to find this in docs and couldn't for some reason. This is exactly what I want, thanks